esp-hal/src/lib.rs

```diff
@@ -745,12 +745,42 @@ use crate::{
     peripherals::Peripherals,
 };
 
+/// Watchdog status.
+#[derive(Default, PartialEq)]
+pub enum WatchdogStatus {
+    /// Enables a watchdog timer with the specified timeout.
+    Enabled(fugit::MicrosDurationU64),
```

Comment:
There's potentially further configuration for WDTs which a user may which to provide, such as clock source and timeout action. Are these things we want to consider here, or will those be set via separate APIs? If we want to configure that all here as well in the future then we should perhaps change this to a struct.

Reply:
Hmm, I think there are two paths here:

* We keep it as is for a basic enable, and if a user wants to do more advanced things they can configure it via the peripheral driver itself later.
* We remove these configs in favour of a global `DisableWatchDogs` flag which will simply disable any enabled watchdogs from the bootloader or whatever.

I personally don't have a strong opinion, I think they both make some level of sense, so I'd be interested what you/others think.

Reply:
I haven't thought of this option but now I think I'd prefer

> We keep it as is for a basic enable, and if a user wants to do more advanced things they can configure it via the peripheral driver itself later.

This will give the user all the freedom they want if needed but makes simple things easy (But also not a strong opinion)